pandas/core/indexes/multi.py

```diff
@@ -1162,6 +1162,11 @@ def from_arrays(cls, arrays, sortorder=None, names=None):
         MultiIndex.from_product : Make a MultiIndex from cartesian product
                                   of iterables
         """
+        if not is_list_like(arrays):
+            raise TypeError("Inappropriate input for list of arrays.")
```

Comment:
make a more user-friendly message, maybe

``input must be a list-like of arrays``